File: question5.js
class LegendaryDrum {
    constructor(material, size) {
      this.material = material;
      this.size = size;
    }
    playSound(tone) {
    //   console.log(`The drum produces ${tone} sound`);
      console.log(`The ${this.constructor.name} drum is made of ${this.material} and is of size ${this.size}`);
    }
  }
  class Djembe extends LegendaryDrum {
    // constructor(material,size)
    // super(material,size)
    playSound(tone) {
      super.playSound(tone);
      console.log(`The Djembe drum produces ${tone} sound`);
    }
  }
  class TalkingDrum extends LegendaryDrum {
    playSound(tone) {
      super.playSound(tone);
      console.log(`The Talking Drum produces ${tone} sound`);
    }
  }
  class Bouragarabou extends LegendaryDrum {
    playSound(tone) {
      super.playSound(tone);
      console.log(`The Bouragarabou drum produces ${tone} sound`);
    }
  }
  const myTalkingDrum = new TalkingDrum("wood", "large");
  myTalkingDrum.playSound("deep");
  const myDjembe  = new Djembe ("leather", "large");
  myDjembe.playSound("dududu");
  const myBouragarabou = new Bouragarabou("plastic", "small");
  myBouragarabou.playSound("low-high");
  
  

File: pythonperspective/question5.py
class Druming:
    def __init__(self, material, size):
        self.material = material
        self.size = size
    def playSounds(self, tones):
        print(f"This {self.size} {type(self).__name__} drum is made of {self.material} and produces {tones} tone")

class Djembe(Druming):
    def playSounds(self, tones):
        super().playSounds(tones)

class TalkingDruming(Druming):
    def __init__(self, material, size, height):
        super().__init__(material, size)
        self.height = height
    def playSounds(self, tones):
        # super().playSounds(tones)
        print(f"This {self.size} {type(self).__name__} mimics the lines of human {tones} speech and has {self.height}")
        return self.height
class Bougarabous(Druming):
    def playSounds(self, tones):
        super().playSounds(tones)

drum = []
drum.append(Djembe("wood", "Small"))

drum.append(Bougarabous("plastic", "medium"))

theDjembeing = Djembe("wood", "small")
theDjembeing.playSounds("heavy")
theTalkingDrum = TalkingDruming("leather", "huge", "3 feet")
theTalkingDrum.playSounds("crunch crunch")
theBougarabous = Bougarabous("animalskin", "big")
theBougarabous.playSounds("deep")






